ledger/block/src/transactions/confirmed/mod.rs

```diff
@@ -51,20 +51,22 @@ impl<N: Network> ConfirmedTransaction<N> {
             }
         };
 
-        // Count the number of `InitializeMapping` and `UpdateKeyValue` finalize operations.
-        let (num_initialize_mappings, num_update_key_values) =
-            finalize_operations.iter().try_fold((0, 0), |(init, update), operation| match operation {
-                FinalizeOperation::InitializeMapping(..) => Ok((init + 1, update)),
-                FinalizeOperation::UpdateKeyValue(..) => Ok((init, update + 1)),
+        // Count the number of `InitializeMapping` and `*KeyValue` finalize operations.
+        let (num_initialize_mappings, num_key_values) =
+            finalize_operations.iter().try_fold((0, 0), |(init, key_value), operation| match operation {
+                FinalizeOperation::InitializeMapping(..) => Ok((init + 1, key_value)),
+                FinalizeOperation::InsertKeyValue(..) // At the time of writing, `InsertKeyValue` is only used in tests. However, it is added for completeness, as it is a valid operation.
+                | FinalizeOperation::RemoveKeyValue(..)
```

Comment:
Just for the record, this may be another theoretical adjustment to transaction validity.

Reply:
Are there any checks we should put in place here?

Reply:
On additional thought, no need for checks, but can you document in the docblocks of `{accepted, rejected}_{deploy, execute}` whether `finalize_operations` contains e.g. operations from `Fee`, `constructor`, etc.